tools/pipelines/templates/include-test-perf-benchmarks-install-package.yml

```diff
@@ -0,0 +1,84 @@
+# Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+# Licensed under the MIT License.
+
+# This template is intended to be used for common steps across stages of the test-perf-benchmarks.yml pipeline.
+# It downloads (from the artifacts produced by another pipeline) and installs a specified package that has performance tests.
+
+parameters:
+# Identifier for the pipeline that produced the artifact with the package to be installed.
+# Will be used for the 'pipeline' input for a DownloadPipelineArtifact task.
+- name: artifactPipeline
+  type: string
+
+# Identifier for the pipeline run that produced the artifact with the package to be installed.
+# Will be used for the 'buildId' input for a DownloadPipelineArtifact task.
+- name: artifactBuildId
+  type: string
+
+# Name of the package to be installed.
+- name: testPackageName
+  type: string
+
+# Path where the package should be installed. This template assumes an appropriate .npmrc file was already setup there
+# so dependencies can be resolved from our feeds.
+- name: installPath
+  type: string
+
+steps:
+- task: Bash@3
+  displayName: Set local template variables
+  inputs:
+    targetType: 'inline'
+    # Using a prefix on these variables to try to make them unique to minimize risk of conflicts with variables that
+    # use the same name in the pipeline that includes this template.
+    # Using isOutput=true variables was too complicated/dirty.
+    script: |
+      echo "Setting local variables for yml template"
+
+      # Doing the character replacements with sed at runtime because using replace() in ADO template expression (which
+      # are evaluated at compile-time) causes some weird interactions if the value provided for testPackageName is only
+      # available at runtime. Also, using ! as separator instead of the usual / because forward slash is something we
+      # need to replace.
+      TEST_PACKAGE_NAME=$(echo "${{ parameters.testPackageName }}" | sed -r 's!@!!g' | sed -r 's!/!-!g')
+      echo "##vso[task.setvariable variable=itpbip_sanitizedPackageName]${TEST_PACKAGE_NAME}-?.?.?-*.tgz"
+      echo "##vso[task.setvariable variable=itpbip_downloadPath]$(Pipeline.Workspace)/downloadedPackages"
```

Comment:
Character replacements here are now done at runtime. The previous approach using template expression syntax like `${{ replace( ... ) }}` has issues if the value passed to the parameter is a runtime variable like `$(testPackage)`, because the replacements of `@` and `/` occurs at compile time and thus _on the string '$(testPackage)' itself_, not on its eventual runtime value. A workaround was to pass that particular parameter with template expression syntax like `${{ testPackage }}`, which was possible in this case, but I thought it could become confusing in the future and is not guaranteed to be possible in all cases, so doing the replacements at runtime makes the template more robust and easier to use.